python/dolfinx/fem/function.py

```diff
@@ -473,79 +478,112 @@ def split(self) -> tuple[Function, ...]:
 
     def collapse(self) -> Function:
         u_collapsed = self._cpp_object.collapse()
-        V_collapsed = FunctionSpace(self.function_space._mesh, self.ufl_element(), u_collapsed.function_space)
-        return Function(V_collapsed, Vector(u_collapsed.x))
+        V_collapsed = FunctionSpaceBase(self.function_space._mesh, self.ufl_element(), u_collapsed.function_space)
+        return Function(V_collapsed, la.Vector(u_collapsed.x))
 
 
 class ElementMetaData(typing.NamedTuple):
-    """Data for representing a finite element"""
+    """Data for representing a finite element
+
+    :param family: Element type.
+    :param degree: Polynomial degree of the element.
+    :param shape: Shape for vector/tensor valued elements that are
+        constructed from blocked scalar elements (e.g., Lagrange).
+    :param symmetry: Symmetry option for blocked tensor elements.
+
+    """
     family: str
     degree: int
+    shape: typing.Optional[typing.Tuple[int, ...]] = None
+    symmetry: typing.Optional[bool] = None
+
+
+def FunctionSpace(mesh: Mesh,
+                  element: typing.Union[ufl.FiniteElementBase, ElementMetaData,
+                                        typing.Tuple[str, int, typing.Tuple, bool]],
+                  form_compiler_options: typing.Optional[dict[str, typing.Any]] = None,
+                  jit_options: typing.Optional[dict[str, typing.Any]] = None) -> FunctionSpaceBase:
+    """Create a finite element function space.
+
+    Args:
+        mesh: Mesh that space is defined on
+        element: Finite element description
+        form_compiler_options: Options passed to the form compiler
+        jit_options: Options controlling just-in-time compilation
+
+    Returns:
+        A function space.
+
+    """
+
+    # Create UFL element
+    try:
+        e = ElementMetaData(*element)
+        ufl_e = basix.ufl.element(e.family, mesh.basix_cell(), e.degree, shape=e.shape,
+                                  gdim=mesh.ufl_cell().geometric_dimension())
+    except TypeError:
+        ufl_e = element  # type: ignore
```

Comment:
The symmetry option doesn't seem to be passed down from the `ElementMetaData` to basix

Reply:
Fixed.